File: DSA/Hima-Swarupa-Padala/day10q2.py
class QueueUsingStacks:
    def __init__(self):
        self.stack1 = []  
        self.stack2 = []  

    def enqueue(self, item):
        self.stack1.append(item)

    def dequeue(self):
        if not self.stack2:
            while self.stack1:
                self.stack2.append(self.stack1.pop())

        if self.stack2:
            return self.stack2.pop()
        else:
            raise IndexError("Queue is empty")

my_queue = QueueUsingStacks()
my_queue.enqueue(1)
my_queue.enqueue(2)
print(my_queue.dequeue())  
print(my_queue.dequeue())  

File: DSA/Shruti-Pandey/Day_10_q2.java
import java.util.Stack;

class QueueUsingStacks<T> {
    private Stack<T> stack1;
    private Stack<T> stack2;

    public QueueUsingStacks() {
        stack1 = new Stack<>();
        stack2 = new Stack<>();
    }

    public void enqueue(T item) {
        stack1.push(item);
    }

    public T dequeue() {
        if (isEmpty()) {
            throw new IllegalStateException("Queue is empty");
        }
        
        if (stack2.isEmpty()) {
            while (!stack1.isEmpty()) {
                stack2.push(stack1.pop());
            }
        }
        
        return stack2.pop();
    }

    public boolean isEmpty() {
        return stack1.isEmpty() && stack2.isEmpty();
    }

    public static void main(String[] args) {
        QueueUsingStacks<Integer> queue = new QueueUsingStacks<>();
        
        queue.enqueue(1);
        queue.enqueue(2);
        
        System.out.println(queue.dequeue()); 
        System.out.println(queue.dequeue()); 
    }
}
